flyteadmin/pkg/repositories/transformers/task_execution.go

```diff
@@ -437,6 +467,11 @@ func UpdateTaskExecutionModel(ctx context.Context, request *admin.TaskExecutionE
 		return errors.NewFlyteAdminErrorf(codes.Internal, "failed to merge task event custom_info with error: %v", err)
 	}
 	taskExecutionClosure.Metadata = mergeMetadata(taskExecutionClosure.Metadata, request.Event.Metadata)
+
+	if isPhaseChange {
+		filterExternalResourceLogsByPhase(taskExecutionClosure.Metadata.ExternalResources, request.Event.Phase)
+	}
+
```

Comment:
This is required map tasks which treat log links differently for performance reasons.

Master:

<img width="703" alt="map_master" src="https://github.com/flyteorg/flyte/assets/36511035/09a25c6c-6b4f-461a-980b-d5a9c30b352d">

This branch:

<img width="639" alt="map_branch" src="https://github.com/flyteorg/flyte/assets/36511035/cb42e881-b952-40c3-8ac5-ee17817f1cb1">

Note that the task execution view with the log links only becomes available in the running phase for map tasks, meaning that there is no use case for showing log links already in the pending phase. However, log links can be removed once finished.